crates/swc_ecma_transforms_compat/tests/optional-chaining-private-var/issue-7561/output.js

```diff
@@ -0,0 +1,10 @@
+// …
+var _x = new WeakMap();
+class Foo {
+    test() {
+        (_this = this) === null || _this === void 0
+          ? void 0
+          : _class_private_field_get(_this.y , _x);
```

Comment:
This is wrong.

```js
class Foo {
  #x;
  test() {
    this === null || this === void 0 ? void 0 : this.y.#x;
  }
}
```

Please try locally using 

```json
{
  "plugins": [
    "@babel/plugin-transform-optional-chaining"
  ]
}
```

This is how I got the expected output

Reply:
Hum, but on the issue: https://github.com/swc-project/swc/issues/7561, @jridgewell says that this is the expected output:

```ts
// …
var _x = new WeakMap();
class Foo {
    test() {
        (_this = this) === null || _this === void 0
          ? void 0
          : _class_private_field_get(_this.y , _x);
    }
    // …
}
```

Also, your output is not transforming the private field. Shouldn't it be transformed as well?

Reply:
Hey @kdy1, 

On the original issue: https://github.com/swc-project/swc/issues/7561, @jridgewell says that this is the expected output:

```ts
// …
var _x = new WeakMap();
class Foo {
    test() {
        (_this = this) === null || _this === void 0
          ? void 0
          : _class_private_field_get(_this.y , _x);
    }
    // …
}
```

Also, isn't your output the same as mine, but without the transformation for the private field?

Reply:
PS: I thought that the main problem was that we were not transforming the private field.

Reply:
No. Expected output of `optional-chaining` is mine. What you uploaded is output of full transforms.

Reply:
I'm sorry, @kdy1, I don't think I understand your comment.

Here is the problem that I want to solve with this PR (please note that I haven't been able to actually solve it yet, this is why you only saw a test): whenever we have an optional chaining expression with a private field, the private field is not transformed (we don't replace it with: `_class_private_field_get`).

This is what I understood from these issues: https://github.com/swc-project/swc/issues/7561 and https://github.com/swc-project/swc/issues/7559.

Regarding your comment:

> No. Expected output of optional-chaining is mine. What you uploaded is output of full transforms.

Do you mean that I've put the test in the wrong place? Since it is inside the optional chaining module, it shouldn't be testing private fields? If so, where would be the most appropriate place to put it?

Or do you mean that I am solving the wrong problem?

Reply:
> whenever we have an optional chaining expression with a private field, the private field is not transformed

This is wrong. Private fields are transformed by another pass.

Reply:
The optional chaining pass should produce

```js
class Foo {
  #x;
  test() {
    this === null || this === void 0 ? void 0 : this.y.#x;
  }
}
```

and another pass is responsible for the rest.

Reply:
Ok, there are two problems?

One for the optional chaining and another for tge private field?

Reply:
No, private field transforms are working correctly

Reply:
I've updated the test as you requested.